### PR TITLE
feat: use new input component everywhere

### PR DESCRIPTION
**Pull request by MounirDhahri**
This PR resolves [] 

### Description

This PR adds the new input component everywhere in Eigen.
TODO: Add masking for phone numbers, it was never working properly and this PR exposed it to me.

**Please note:** Once the Input changes are approved - I will update everything to use **Input** instead of **Input2**. I am doing this intermediate step because I couldn't find any better way to test both out gradually.

Related Palette changes https://github.com/artsy/palette-mobile/pull/207

Group 11 (1)

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- use new input component everywhere  - mounir

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (use new input component everywhere  - mounir)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against eb6291b31666a94c97980779b0394f6dd967f518
</p>

File: package.json [hunk 1]
```diff
@@ -108,7 +108,7 @@
   },
   "dependencies": {
     "@artsy/cohesion": "4.170.0",
-    "@artsy/palette-mobile": "13.1.27",
+    "@artsy/palette-mobile": "13.1.28--canary.207.1767.0",
```
Comment (olerichter00) [inline package.json:111]: 👀

File: src/app/Components/Artist/ArtistInsights/MarketStats.tsx [hunk 2]
```diff
@@ -128,8 +128,8 @@ const MarketStats: React.FC<MarketStatsProps> = ({ priceInsightsConnection }) =>
       <Select
         value={selectedPriceInsight.medium}
         options={mediumOptions.current}
-        title="Select medium"
-        showTitleLabel={false}
+        title=""
```
Comment (olerichter00) [inline src/app/Components/Artist/ArtistInsights/MarketStats.tsx:131]: Can we remove the title prop entirely? If not, would be great to make the prop optional.

```suggestion
```

File: src/app/Components/Bidding/Screens/SelectMaxBid.tsx [hunk 3]
```diff
@@ -157,13 +156,18 @@ export const SelectMaxBidQueryRenderer: React.FC<{
...
+          if (props.artwork?.sale_artwork && props.me) {
+            return (
+              <SelectMaxBidContainer
+                me={props.me}
+                sale_artwork={props.artwork.sale_artwork}
+                navigator={navigator}
+              />
+            )
+          }
+          return null
```
Comment (olerichter00) [inline src/app/Components/Bidding/Screens/SelectMaxBid.tsx:169]: https://medium.com/swlh/return-early-pattern-3d18a41bba8 🤷 

```suggestion
          if (!props.artwork?.sale_artwork || !props.me) {
            return null
          }
            return (
              <SelectMaxBidContainer
                me={props.me}
                sale_artwork={props.artwork.sale_artwork}
                navigator={navigator}
              />
            )
```

File: src/app/Scenes/MyAccount/MyAccountEditPassword.tsx [hunk 4]
```diff
@@ -109,19 +107,21 @@ export const MyAccountEditPassword: React.FC<{}> = ({}) => {
       </Flex>
       <Separator mb={2} mt={4} />
       <Stack mx={2}>
-        <Input
-          description="Must include at least one uppercase letter, one lowercase letter, and one number."
+        <Text>
+          Password must include at least one uppercase letter, one lowercase letter, and one number.
+        </Text>
+        <Input2
+          // TODO: Look into this with design
```
Comment (olerichter00) [inline src/app/Scenes/MyAccount/MyAccountEditPassword.tsx:114]: 👀

**Review by olerichter00 — COMMENTED** [review]
Phew! There is a lot going on here! I've tried to go through all the changes and test all relevant screens, but these big PRs are pretty hard to review.

It might be reasonable to make the switch to the new input component in one step, but if possible, I would suggest making changes like this in multiple steps. Maybe using something like an adapter or not updating all screens at once. This approach might not be ideal but should be acceptable as well.

The code looks great! 🌟 

Testing the changes on the simulator, I've got a few error messages:

When clicking "Add Artists" on My Collection screen:
Simulator Screenshot - iPhone 15 Pro - 2024-04-17 at 11 39 51

---

When clicking "Add Artworks" on My Collection screen:

Simulator Screenshot - iPhone 15 Pro - 2024-04-17 at 11 40 56

--- 

When creating an alert from the artwork screen:

Simulator Screenshot - iPhone 15 Pro - 2024-04-17 at 11 41 51

File: src/app/Components/Bidding/Screens/SelectMaxBid.tsx (hunk 3 above)
Comment (MounirDhahri) [reply src/app/Components/Bidding/Screens/SelectMaxBid.tsx:169]: good note, I copied this code and didn't notice this

File: src/app/Components/Artist/ArtistInsights/MarketStats.tsx (hunk 2 above)
Comment (MounirDhahri) [reply src/app/Components/Artist/ArtistInsights/MarketStats.tsx:131]: will do it

File: package.json (hunk 1 above)
Comment (MounirDhahri) [reply package.json]: sorry about these 😄 I was still getting working on it

**Comment by MounirDhahri** [comment]
Hey @olerichter00 Good point, I think the main issue here was that the changes need to be reflected everywhere if we want to make sure that we have the same input design across the app. 
My rollout strategy with this will be different though, I am planning an out of schedule QA session with Applause today in order to make sure everything is alright, meanwhile, this PR will keep pointing to a canary build from palette-mobile and won't be merged unless everything has been approved. 
About the errors you had, they should be gone now - maybe you pulled at a time when the PR wasn't working. I will mark the PR as ready for review although I didn't make much changes since you reviewed it other than renaming back to the `Input` (instead of `Input2`)

**Pull request merged**